consistency/tool/comparison/comparison.py

```diff
@@ -1,42 +1,37 @@
 from strsimpy.sorensen_dice import SorensenDice
-import sys
+from google.cloud.apigeeregistry.applications.v1alpha1.consistency import (
+    consistency_report_pb2 as cr,
+)
 
 
 class Comparison:
-    def __init__(self, stub, new_words):
+    def __init__(self, stub, new_words, word_groups=None, noise_words_index = None):
         self.stub = stub
         self.new_words = new_words
-
-    def get_word_group_ids(self):
-        # TODO: retrieve the WordGroups we have stored from registry.
-        # return them in an appropriate format for easy comparison
-        return []
+        self.word_groups=word_groups
+        self.noise_words_index  = noise_words_index 
 
     # format: {word : (cluster_id, distance)}
-    def find_word_groups(self):
-        try:
-            ids = self.get_word_group_ids()
-        except Exception as e:
-            print(e, "Getting word group IDs failed.")
+    def find_closest_word_groups(self):
 
         # the maximum possible score for dice distance is 1, signifying complete dissimilarity.
         if (
-            ids == None
-            or len(ids) < 1
+            self.word_groups == None
+            or len(self.word_groups) < 1
             or self.new_words == None
             or len(self.new_words) < 1
         ):
             return None
 
         def find_closest_id(word):
             dice = SorensenDice(2)
-            comparsion_info = ["UNIQUE_WORD", 1]
-            for id in ids:
-                distance = dice.distance(word, id)
+            comparsion_info = ["POSSIBLE_UNIQUE_WORD", 1]
```

Comment:
Can we initialize `comparison_info` with `[noise_word_group, 1]`. If no cluster is found for a term, that term will possibly exist in the noise_word_group.